code/obj/flock/structure/fabricator.dm

```diff
@@ -0,0 +1,77 @@
+/obj/flock_structure/fabricator
+	icon_state = "reclaimer" // placeholder
+	name = "angled pedestal"
+	desc = "A strange machine. It appears to have some sort of output slot?"
+	flock_desc = "A converter that turns its contents into resource cubes."
+	flock_id = "Fabricator"
+	health = 20
+	health_max = 20
+	repair_per_resource = 1
+	passthrough = TRUE
+
+	var/resources_to_produce = 0
+	var/done = FALSE
+
+	New(atom/location, datum/flock/F = null, obj/content_holder)
+		..()
+		if (istype(content_holder, /obj/machinery/vending))
+			var/obj/machinery/vending/vending_machine = content_holder
+			for (var/datum/data/vending_product/product as anything in vending_machine.product_list)
+				src.resources_to_produce += get_initial_item_health(product.product_path) * product.product_amount
+		else if (istype(content_holder, /obj/machinery/manufacturer))
+			var/obj/machinery/manufacturer/fab = content_holder
+			if (fab.free_resource_amt)
```

Comment:
iirc, `free_resource_amt` is added to the fab's resource pool on initialisation if it's created  by map and not player. I think what you want here is just `fab.contents`?

Reply:
free_resource_amt is used initially and then resources are claimed and it uses contents